### ISSUE TITLE
[MemberExpression] Brackets are required when calling a method on a number

### ISSUE THREAD

**Issue by karl** [jlongster/prettier#96]
Really excited about Prettier! I just ran it against our codebase here at Geckoboard and discovered a few edge case bugs. I'll open issues for each of the bugs that I can generate a small repro case for.

You can call methods on a number using the following syntax:

```
(0).toLocaleString();
```

Prettier currently converts this to:

```
0.toLocaleString();
```

Which gives the following syntax error when run in Chrome:

```
Uncaught SyntaxError: Invalid or unexpected token
```

I would have expected the output to be:

```
(0).toLocaleString();
```

https://vjeux.github.io/prettier-browser/#%7B%22content%22%3A%22(0).toLocaleString()%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22useFlowParser%22%3Afalse%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Afalse%7D%7D

**Comment by chabou** [comment]
Same bug here:
`return (((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1);`
is wrongly converted to:
`return (1 + Math.random()) * 65536 | 0.toString(16).substring(1);`

https://vjeux.github.io/prettier-browser/#%7B%22content%22%3A%22(((1%20%2B%20Math.random())%20*%200x10000)%20%7C%200).toString(16).substring(1)%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22useFlowParser%22%3Afalse%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Afalse%7D%7D

**Comment by jlongster** [comment]
> Same bug here:
> return (((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1);
> is wrongly converted to:
> return (1 + Math.random()) * 65536 | 0.toString(16).substring(1);

What's also disconcerting is `0x10000` is converted to `65536`. I hope the AST has enough info for us to avoid converting the number.

**Comment by vjeux** [comment]
The first issue has been fixed, but 

```js
(((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1);
```

is still broken

**Comment by vjeux** [comment]
```js
(1 | 0).toString(16);
> (1 | 0).toString(16); // Good

(1 | 0).toString(16).substring(1);
> 1 | 0.toString(16).substring(1); // Bad
```

Interestingly, this only happens when there are two chained calls

https://jlongster.github.io/prettier/#%7B%22content%22%3A%22(1%20%7C%200).toString(16)%3B%5Cn(1%20%7C%200).toString(16).substring(1)%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Afalse%7D%7D

**Comment by vjeux** [comment]
`needsParens` is not being called for the binary expression. I think that it's because `printMemberChain` somehow skips it but I don't understand enough of how it works to be sure.

**Comment by vjeux** [comment]
This is now fixed in 0.0.9
